### ISSUE TITLE
How do I define dependency architecture?

### ISSUE THREAD

**Issue by bviktor** [jordansissel/fpm#1736]
I'm build an x64 RPM for Arm Development Studio, which depends on a couple of i686 packages. I specified those dependencies in the fpm command line as:

```
-d gtk2 -d webkitgtk3 -d mesa-libGLU -d glibc.i686 -d libgcc.i686 -d libstdc++.i686 -d zlib.i686
```

Unfortunately that doesn't work:

```
# yum install arm-ds-2020.0
Loaded plugins: nvidia, product-id, search-disabled-repos, subscription-manager
Resolving Dependencies
--> Running transaction check
---> Package arm-ds-2020.0.x86_64 0:1-1 will be installed
--> Processing Dependency: zlib.i686 for package: arm-ds-2020.0-1-1.x86_64
--> Processing Dependency: libstdc++.i686 for package: arm-ds-2020.0-1-1.x86_64
--> Processing Dependency: libgcc.i686 for package: arm-ds-2020.0-1-1.x86_64
--> Processing Dependency: glibc.i686 for package: arm-ds-2020.0-1-1.x86_64
--> Finished Dependency Resolution
Error: Package: arm-ds-2020.0-1-1.x86_64 (fpga)
           Requires: libgcc.i686
Error: Package: arm-ds-2020.0-1-1.x86_64 (fpga)
           Requires: libstdc++.i686
Error: Package: arm-ds-2020.0-1-1.x86_64 (fpga)
           Requires: zlib.i686
Error: Package: arm-ds-2020.0-1-1.x86_64 (fpga)
           Requires: glibc.i686
**********************************************************************
yum can be configured to try to resolve such errors by temporarily enabling
disabled repos and searching for missing dependencies.
To enable this functionality please set 'notify_only=0' in /etc/yum/pluginconf.d/search-disabled-repos.conf
**********************************************************************

Error: Package: arm-ds-2020.0-1-1.x86_64 (fpga)
           Requires: libgcc.i686
Error: Package: arm-ds-2020.0-1-1.x86_64 (fpga)
           Requires: libstdc++.i686
Error: Package: arm-ds-2020.0-1-1.x86_64 (fpga)
           Requires: zlib.i686
Error: Package: arm-ds-2020.0-1-1.x86_64 (fpga)
           Requires: glibc.i686
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
```

Even though:

```
# yum install libgcc.i686 libstdc++.i686 zlib.i686 glibc.i686
Loaded plugins: nvidia, product-id, search-disabled-repos, subscription-manager
Package libgcc-4.8.5-44.el7.i686 already installed and latest version
Package libstdc++-4.8.5-44.el7.i686 already installed and latest version
Package zlib-1.2.7-18.el7.i686 already installed and latest version
Package glibc-2.17-317.el7.i686 already installed and latest version
Nothing to do
```

So my guess is that the RPM generated by FPM wants to install literally the 'zlib.i686' package, instead of the i686 version of zlib.

**Comment by bviktor** [comment]
Here's the answer:

https://unix.stackexchange.com/a/52456/88217

TLDR use the `(x86-32)` suffix instead of `.i686`. So in my case, the erroneous

```
-d gtk2 -d webkitgtk3 -d mesa-libGLU -d glibc.i686 -d libgcc.i686 -d libstdc++.i686 -d zlib.i686
```

becomes

```
-d gtk2 -d webkitgtk3 -d mesa-libGLU -d glibc(x86-32) -d libgcc(x86-32) -d libstdc++(x86-32) -d zlib(x86-32)
```
